### ISSUE TITLE
Composer doesn't work with local filesystem subversion repos

### ISSUE THREAD

**Issue by glennschmidt** [composer/composer#5596]
When an svn repository is referenced with a `file://` URL, or a local pathname, Composer cannot download packages from the repository.
eg.
```json
    "repositories": [
        {
            "type": "svn",
            "url": "file:///srv/repos/mine",
```

This happens because Composer erroneously passes file:// URLs through `realpath()`, which will return `false` because it expects a path, not a URL. For example, in `Composer\Downloader\VcsDownloader::download()`
```php
if (Filesystem::isLocalPath($url)) {
    $url = realpath($url);
}
```

`Filesystem:isLocalPath()` returns `true` for file:// URLs, but `realpath` only accepts pathnames.

Also, even if you refer to the local repository in `composer.json` using a pathname (`/srv/repos/mine` instead of `file:///srv/repos/mine`), `SvnDriver::normalizeUrl()` turns it into a file:// URL, and you still get this problem.

**Comment by alcohol** [comment]
Did you try using the VCS driver?

**Comment by Seldaek** [comment]
As a workaround you can probably just use the path `/srv/repos/mine` without file://. Otherwise send a pull request to fix it so that file:// is stripped before doing a realpath.

**Comment by Seldaek** [comment]
Fixed by #5688
